deploy/pre-deployment/pre-deployment-check.sh

```
@@ -175,6 +204,113 @@ check_cluster_resources() {
     return 1
 }
 
+check_intel_gpu_resources() {
+    print_section "Checking cluster Intel GPU resources"
+
+    # Check for gpu.intel.com DeviceClass
+    if kubectl get deviceclass gpu.intel.com &> /dev/null; then
+        print_status $GREEN "✅ DeviceClass gpu.intel.com found"
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)